src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
+
+enum ItemsPerPage {
+  three = 3,
+  five = 5,
+  ten = 10,
+  twenty = 20,
+}
 
 export const App: React.FC = () => {
+  const [, setSearchParams] = useSearchParams();
+  const [perPage, setPerPage] = useState(ItemsPerPage.five);
+  const [currentPage, onPageChange] = useState(1);
```

Comment:
The naming of the state setter function is a bit misleading. Typically, setter functions are named after the state variable they are meant to set. In this case, `setCurrentPage` would be more appropriate than `onPageChange`, as the latter sounds like an event handler rather than a state setter.